pkg/ovsdb/monitor_test.go

```diff
@@ -1,97 +1,242 @@
+			expTable: "table", expUuid: "uuid",
+			event: clientv3.Event{Type: mvccpb.PUT,
+				PrevKv: &mvccpb.KeyValue{Key: []byte("key/db/table/uuid"), Value: []byte("{\"c1\":\"v1\",\"c2\":\"v2\"}")},
+				Kv:     &mvccpb.KeyValue{Key: []byte("key/db/table/uuid"), Value: []byte("{\"c1\":\"v1\",\"c2\":\"v3\"}"), CreateRevision: 1, ModRevision: 2}},
+			expRowUpdate: nil},
 	}
-	ch := NewHandler(context.Background(), dbServer)
-	cmp := ovsjson.CondMonitorParameters{}
-	cmp.DatabaseName = dataBase
-	cmp.JsonValue = nil
-	mcr := ovsjson.MonitorCondRequest{}
-	for _, sel := range selects {
-		for _, col := range columns {
-			mcr.Columns = col
-			mcr.Select = sel
-			cmp.MonitorCondRequests = map[string][]ovsjson.MonitorCondRequest{tableName: {mcr}}
-
-			monitorResponse, actualError := ch.Monitor(nil, cmp)
-			assert.Nilf(t, actualError, "handler call should not return error: %v", err)
-
-			monitorCondResponse, actualError := ch.MonitorCond(nil, cmp)
-			assert.Nilf(t, actualError, "handler call should not return error: %v", err)
-
-			monitorSinceResponse, actualError := ch.MonitorCondSince(nil, cmp)
-			assert.Nilf(t, actualError, "handler call should not return error: %v", err)
-
-			testResponse := func(resp ovsjson.TableUpdates, isV1 bool) {
-				if sel != nil && !sel.Initial {
-					assert.Truef(t, len(resp) == 0, "If Select.Initial is `false` monitor requests should not return data")
-					return
-				}
-				lr, ok := resp[tableName]
-				assert.True(t, ok)
-				assert.True(t, len(lr) == len(*values))
-				for name := range *values {
-					lr1, ok := lr[name]
-					assert.True(t, ok)
-					if len(col) == 0 {
-						col = allColumns
+	for name, ts := range tests {
+		tableName, uuid, row, err := ts.updater.prepareTableUpdate(&ts.event)
+		if ts.err != nil {
+			assert.EqualErrorf(t, err, ts.err.Error(), "[%s test] expected error %s, got %v", name, ts.err.Error(), err)
+			continue
+		} else {
+			assert.Nilf(t, err, "[%s test] returned unexpected error %v", name, err)
+		}
+		assert.Equalf(t, ts.expTable, tableName, "[%s test] returned wrong table name, expected %q, got %q", name, ts.expTable, tableName)
+		assert.Equalf(t, ts.expUuid, uuid, "[%s test] returned wrong row uuid, expected %q, got %q", name, ts.expUuid, uuid)
+		if ts.expRowUpdate == nil {
+			assert.Nilf(t, row, "[%s test] returned unexpected row %#v", name, row)
+		} else {
+			assert.NotNil(t, row, "[%s test] returned nil row", name)
+			if ts.updater.isV1 {
+				ok, msg := row.ValidateRowUpdate()
+				assert.Truef(t, ok, "[%s test]  Row update is not valid %s %#v", name, msg, row)
+			} else {
+				ok, msg := row.ValidateRowUpdate2()
+				assert.Truef(t, ok, "[%s test]  Row update is not valid %s %#v", name, msg, row)
+			}
+
+			assert.EqualValuesf(t, ts.expRowUpdate, row, "[%s test] returned wrong row update, expected %#v, got %#v", name, *ts.expRowUpdate, *row)
+		}
+
+	}
+
+}
+
+func TestMonitorResponse(t *testing.T) {
+	/*
```

Comment:
you really don't want to commit this ... if not in use then delete the content

Reply:
let me think about this test more, I don't want to remove it for now.